src/replication.c

```diff
@@ -2526,6 +3152,143 @@ int replicaTryPartialResynchronization(connection *conn, int read_reply) {
     return PSYNC_NOT_SUPPORTED;
 }
 
+/* Replication: Replica side.
+ * This connection handler fires after rdb-connection was initialized. We use it
+ * to adjust the replica main for loading incremental changes into the local buffer. */
+void setupMainConnForPsync(connection *conn) {
+    int psync_result = -1;
+    char llstr[LONG_STR_SIZE];
+    char *err = NULL;
+    if (server.repl_state == REPL_STATE_SEND_HANDSHAKE) {
+        /* We already have an initialized connection at primary side, we only need to associate it with RDB connection */
+        ll2string(llstr, sizeof(llstr), server.rdb_client_id);
+        err = sendCommand(conn, "REPLCONF", "set-rdb-client-id", llstr, NULL);
+        if (err) goto error;
+        server.repl_state = REPL_STATE_RECEIVE_CAPA_REPLY;
+        return;
+    }
+
+    if (server.repl_state == REPL_STATE_RECEIVE_CAPA_REPLY) {
+        err = receiveSynchronousResponse(conn);
+        if (err == NULL) goto error;
+        if (err[0] == '-') {
+            serverLog(LL_NOTICE, "Primary does not understand REPLCONF identify: %s", err);
+            goto error;
+        }
+        sdsfree(err);
+        err = NULL;
+        server.repl_state = REPL_STATE_SEND_PSYNC;
+    }
+
+    if (server.repl_state == REPL_STATE_SEND_PSYNC) {
+        if (server.debug_sleep_after_fork_us) usleep(server.debug_sleep_after_fork_us);
+        if (replicaTryPartialResynchronization(conn, 0) == PSYNC_WRITE_ERROR) {
+            serverLog(LL_WARNING, "Aborting dual connection sync. Write error.");
+            cancelReplicationHandshake(1);
+        }
+        server.repl_state = REPL_STATE_RECEIVE_PSYNC_REPLY;
+        return;
+    }
+    psync_result = replicaTryPartialResynchronization(conn, 1);
+    if (psync_result == PSYNC_WAIT_REPLY) return; /* Try again later... */
+
+    if (psync_result == PSYNC_CONTINUE) {
+        serverLog(LL_NOTICE, "Primary <-> REPLICA sync: Primary accepted a Partial Resynchronization%s",
+                  server.repl_rdb_transfer_s != NULL ? ", RDB load in background." : ".");
+        if (server.supervised_mode == SUPERVISED_SYSTEMD) {
+            serverCommunicateSystemd("STATUS=Primary <-> REPLICA sync: Partial Resynchronization accepted. Ready to "
+                                     "accept connections in read-write mode.\n");
+        }
```

Comment:
any reason for sending this message to systemd? I wasn't expecting one-offs like this but it seems to be the case today. Shouldn't it be wrapped behind `severLog`? curious to hear from @valkey-io/core-team